src/test/python/parser/opentrackio_lib.py

```diff
@@ -8,67 +8,175 @@
+        conversion_factors = {
+            "m": 1.0,
+            "cm": 100.0,
+            "mm": 1000.0,
+            "in": 1000.0 / 25.4,
+        }
+        return conversion_factors[self.value]
+        
+class RotationUnit(Enum):
+    DEGREE = "deg"
+    RADIAN = "rad"
+    
+    def conversion_factor_from_degrees(self) -> float:
+        conversion_factors = {
+            "deg": 1.0,
+            "rad": math.pi/180
+        }
+        return conversion_factors[self.value]
+
+class TimeFormat(Enum):
+    SECONDS = "sec"
+    TIMECODE = "tc"
+    STRING = "string"
+    ISO8601 = "iso8601"
+    
+class FocusDistanceUnit(Enum):
+    METER = "m"
+    CENTIMETER = "cm"
+    MILLIMETER = "mm"
+    INCH = "in"
+    
+    def conversion_factor_from_mm(self) -> float:
+        conversion_factors = {
+            "m": 0.001,
+            "cm": 0.1,
+            "mm": 1.0,
+            "in": 1 / 25.4
+        }    
+        return conversion_factors[self.value]
+        
+class TimeSource(Enum):
+    PTP = "ptp"
+    NTP = "ntp"
+    GENLOCK = "genlock"
+    NONE = "none"
+    
+class PayloadFormat(Enum):
+    JSON = 0x01
+    CBOR = 0x02
+
+def fletcher16(data: bytes) -> bytes:
+    """
+    Compute the Fletcher-16 checksum for the given data.
+    
+    Args:
+        data (bytes): Input data for which the checksum is calculated.
+    
+    Returns:
+        bytes: A 2-byte Fletcher-16 checksum.
+    """
+    sum1 = 0
+    sum2 = 0
+    for byte in data:
+        sum1 = (sum1 + byte) % 255
+        sum2 = (sum2 + sum1) % 255
+    checksum = (sum2 << 8) | sum1
+    return struct.pack('!H', checksum)
+
 
 class OpenTrackIOProtocol:
     """Class to decode and interpret the OpenTrackIO protocol
     Arguments:
-        msg_text: string containing a single json "sample"
         schema_text: string containing a json schema for the protocol
         verbose: Whether to print extra status during processing"""
-    def __init__(self, msg_text=None, schema_text=None, verbose=False):
-        self.sample_str = msg_text              # the Sample raw JSON
+    def __init__(self, schema_text=None, verbose=False):
+        self.sample_str = None              # the Sample raw JSON
         self.schema_str = schema_text           # the Sample raw JSON
         self.verbose = verbose
         self.schema_text = None             # the Schema raw JSON
-        self.trans_mult = 1.0                               # user-preferred units multiplier
-        self.rot_mult = 1.0
+        self.trans_mult = TranslationUnit.METER
+        self.rot_mult = RotationUnit.DEGREE
         self.pd = None          # the parsed procotol dictionary
         self.sd = None          # the parsed schema dictionary
-        self.sample_time_format = "sec"
-        self.sample_time_formats = ["sec","timecode","string"]
+        self.sample_time_format = TimeFormat.ISO8601
         self.focus_dist_mult = 1.0
+        
+        if schema_text:
+            self.import_schema()
```

Comment:
I see now, you're importing it in the constructor. 'Seems fine to do it that way.